plugins/inputs/socket_listener/README.md

```diff
@@ -77,6 +77,40 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
   ## Content encoding for message payloads, can be set to "gzip" to or
   ## "identity" to apply no encoding.
   # content_encoding = "identity"
+
+  ## Packet splitting strategy and corresponding settings.
+  ## Available strategies are:
+  ##   newline         -- split at newlines (default)
+  ##   null            -- split at null bytes
+  ##   delimiter       -- split at delimiter byte-sequence in hex-format
+  ##                      given in `splitting_delimiter`
+  ##   fixed length    -- split after number of bytes given in `splitting_length`
+  ##   variable length -- split depending on length information received in the
+  ##                      data. The length field information is specified in
+  ##                      `splitting_length_field`.
+  # splitting_strategy = "newline"
```

Comment:
#6025 talks about TCP splitting on newline but UDP not splitting. This PR seems to always default to newline. Does this mean it's an incompatible change for UDP, switching from none to newline without a config change?

Reply:
No, the PR only applies the splitting strategy to streams, i.e. TCP (tcp, tcp4, tcp6) and sockets (unix, unixpacket) just as it was before. The default settings reproduces _exactly_ the previous behavior. I maybe should make this clear in the sample.conf documentation.

Reply:
Added a comment to `sample.conf`.